Tella/Scenes/GDrive/ViewModel/GDriveViewModel.swift

```diff
@@ -0,0 +1,87 @@
+//
+
+import Foundation
+import Combine
+class GDriveViewModel: ReportMainViewModel {
+    @Published var selectedReport: GDriveReport?
+    @Published var server: GDriveServer?
+
+    private var delayTime = 0.1
+    
+    var sheetItems : [ListActionSheetItem] { return [
+        
+        ListActionSheetItem(imageName: "view-icon",
+                            content: self.selectedReport?.status.sheetItemTitle ?? "",
+                            type: self.selectedReport?.status.reportActionType ?? .viewSubmitted),
+        ListActionSheetItem(imageName: "delete-icon-white",
+                            content: LocalizableReport.viewModelDelete.localized,
+                            type: ConnectionActionType.delete)
+    ]}
+    
+    init(mainAppModel: MainAppModel) {
+        super.init(mainAppModel: mainAppModel, connectionType: .gDrive, title: LocalizableGDrive.gDriveAppBar.localized)
+        
+        self.getReports()
+        self.getServer()
+        self.listenToUpdates()
+    }
+    
+    private func getServer() {
+        self.server = mainAppModel.tellaData?.gDriveServers.value.first
+    }
+    
+    override func getReports() {
+        getDraftReports()
+        getOutboxedReports()
+        getSubmittedReports()
+    }
+    
+    func getDraftReports() {
+        let draftReports = tellaData?.getDraftGDriveReport() ?? []
+        self.draftReportsViewModel = draftReports.compactMap { report in
+            return ReportCardViewModel(report: report,
+                                       serverName: server?.name,
+                                       deleteReport: { self.deleteReport(report: report) }
+            )
+        }
+    }
+    
+    func getOutboxedReports() {
+        let outboxedReports = tellaData?.getOutboxedGDriveReport() ?? []
+        self.outboxedReportsViewModel = outboxedReports.compactMap { report in
+            ReportCardViewModel(report: report,
+                                serverName: server?.name,
+                                deleteReport: { self.deleteReport(report: report) }
+            )
+        }
+    }
+    
+    func getSubmittedReports() {
+        let submittedReports = tellaData?.getSubmittedGDriveReport() ?? []
+        self.submittedReportsViewModel = submittedReports.compactMap { report in
+            ReportCardViewModel(report: report,
+                                serverName: server?.name,
+                                deleteReport: { self.deleteReport(report: report) }
+            )
+        }
+    }
+    
+    func deleteReport(report: GDriveReport) {
+        let _ = self.mainAppModel.tellaData?.deleteDriveReport(reportId: report.id)
+    }
+    
+    override func listenToUpdates() {
+        self.mainAppModel.tellaData?.shouldReloadGDriveReports
+            .receive(on: DispatchQueue.main)
+            .sink { result in
+            } receiveValue: { draftReports in
+                self.getReports()
+            }.store(in: &subscribers)
+    }
```

Comment:
Maybe we can try to use one method it in the `ReportMainViewModel` (we should think about it )

Reply:
Yeah I agree with this, we should find a way to reuse this method. Let's add it to a next bugbash, thanks for the suggestion rima!

Reply:
https://app.asana.com/0/1199584685142744/1207967764164507 here is the asana task btw, Let's tackle this on the nextcloud release